### ISSUE TITLE
MT_linear_regression still not working

### ISSUE THREAD

**Issue by angerhang** [bibliolytic/MTlearning#2]
After playing around, I found out that for the `MT_linear_regression` class, both the prior and the updated models constantly predict zeros in both my own example and the toy example provided.

Below I tested the prior, updated models and a predictor that predicts only zeros for the toy example in the `testscript`, they yield identical outcomes.
```matlab
>> predictions_prior =  regression_model{i}.prior_predict(T_X2d{5});
    predictions =  new_regression.predict(T_X2d{5});
    zero_predictions = predictions * 0;
    
    sqrt(mean(predictions_prior - T_y{5}).^2)
    sqrt(mean(predictions - T_y{5}).^2)
    sqrt(mean(zero_predictions - T_y{5}).^2)

ans =

    0.5000


ans =

    0.5000


ans =

    0.5000
```

I am suspecting the recent update caused this problem because @atalanti reported that her older version doesn't have this issue. I will have a closer look about the recent changes but any help will be appreciated :D

**Comment by angerhang** [comment]
I tested it on both 2016b and 2014b versions of Matlab, so the version difference shouldn't matter.

**Comment by angerhang** [comment]
This issue has been solved via 9257cb367a879f45d1aa9847e84af2f6646ece7a
